code/Player/Grub/Grub.Death.cs

```diff
@@ -147,6 +149,32 @@ private void FinishDie()
 		}
 	}
 
+	private void IncrementKilledStats()
+	{
+		var attacker = GamemodeSystem.Instance.ActivePlayer;
+		if ( attacker.Client.IsBot ) return;
+
+		string statIdent = "grubs-killed";
+		if ( attacker == Player ) // Killed self.

```

Comment:
The same class that holds the stat names can handle all these increment stat function. It would make more sense if they were not scattered throughout the codebase.